ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -2060,6 +2220,13 @@ modifyTMVar v k =
     atomically (putTMVar v a')
     return b
 
+modifyTMVarSTM :: MonadSTM m => StrictTMVar m a -> (a -> STM m (a, b)) -> STM m b
+modifyTMVarSTM v k = do
+    a <- takeTMVar v
+    (!a',b) <- k a
```

Comment:
```suggestion
    (a', b) <- k a
```
no need for the bang since we use `StrictTVar`.  It's also not needed in other `stm` primitives in this module.  Either a github issue to remove them should be enough.